libqtile/widget/tuned_manager.py

```diff
@@ -0,0 +1,106 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+import re
+import subprocess
+from os import system
+
+from libqtile import bar
+from libqtile.widget import base
+
+
+class TunedManager(base.InLoopPollText):
+    """
+    A widget to interact with the Tuned power management daemon.
+    It always displays the name of the currently active profile.
+
+    The user can define a list of profiles to be used, 3 default
+    profiles exist. These 3 are the default profiles on RHEL and
+    Fedora Linux.
+
+    A left click on the widget will go to the next layout in the
+    list, a right click will go to the previous one.  If the end
+    of the list is reached, it cycles back to the beginning.
+    Scrolling can also be used to cycle through the list, though
+    keep in mind that switching the profile takes a while and so
+    scrolling through the list quickly is not feasible.
+    """
+
+    orientations = base.ORIENTATION_HORIZONTAL
+
+    defaults = [
+        (
+            "modes",
+            ["powersave", "balanced-battery", "throughput-performance"],
+            "The modes to cycle through",
+        )
+    ]
+
+    def __init__(self, **config):
+        base.InLoopPollText.__init__(self, "", **config)
+        self.add_defaults(TunedManager.defaults)
+        self.length_type = bar.CALCULATED
+        self.current_mode = self.find_mode()
+
+        self.add_callbacks(
+            {
+                "Button1": self.next_mode,  # Left Click
+                "Button3": self.previous_mode,  # Right Click
+                "Button4": self.next_mode,  # Scroll up
+                "Button5": self.previous_mode,  # Scroll down
+            }
+        )
+
+    def _configure(self, qtile, bar):
+        base.InLoopPollText._configure(self, qtile, bar)
+        self.text = TunedManager.find_mode(self)
+
+    def poll(self):
+        return self.find_mode()
+
+    def find_mode(self):
+        result = subprocess.run("tuned-adm active", shell=True, capture_output=True, text=True)
+        output = result.stdout
+        regex = re.compile(r"Current active profile:\s+(\S+)")
+        if not regex.findall(output):
+            return ""
+        return regex.findall(output)[0]
```

Comment:
We don't need to call `findall` twice. Use a walruss expression or save the value in a variable before the if statement.